aws/resource_aws_ebs_snapshot_import.go

```diff
@@ -0,0 +1,426 @@
+				if err != nil {
+					return fmt.Errorf("error parsing upload_end to timestamp: %s", err)
+				}
+				client_data.UploadEnd = aws.Time(upload_end)
+			}
+
+			if v, ok := cdv["upload_size"].(float64); ok {
+				client_data.UploadSize = aws.Float64(v)
+			}
+
+			if v, ok := cdv["upload_start"].(string); ok {
+				upload_start, err := time.Parse(time.RFC3339, v)
+				if err != nil {
+					return fmt.Errorf("error parsing upload_start to timestamp: %s", err)
+				}
+				client_data.UploadStart = aws.Time(upload_start)
+			}
+
+			request.ClientData = client_data
+		}
+	}
+
+	request.ClientToken = aws.String(resource.UniqueId())
+
+	if v, ok := d.GetOk("description"); ok {
+		request.Description = aws.String(v.(string))
+	}
+
+	v := d.Get("disk_container")
+	vL := v.([]interface{})
+	for _, v := range vL {
+		dcv := v.(map[string]interface{})
+		disk_container := &ec2.SnapshotDiskContainer{
+			Format: aws.String(dcv["format"].(string)),
+		}
+
+		if v, ok := dcv["description"].(string); ok {
+			disk_container.Description = aws.String(v)
+		}
+
+		if v, ok := dcv["url"].(string); ok && v != "" {
+			disk_container.Url = aws.String(v)
+		}
+
+		if v, ok := dcv["user_bucket"]; ok {
+			vL := v.([]interface{})
+			for _, v := range vL {
+				ub := v.(map[string]interface{})
+				disk_container.UserBucket = &ec2.UserBucket{
+					S3Bucket: aws.String(ub["s3_bucket"].(string)),
+					S3Key:    aws.String(ub["s3_key"].(string)),
+				}
+			}
+		}
+
+		request.DiskContainer = disk_container
+	}
+
+	if v, ok := d.GetOk("encrypted"); ok {
+		request.Encrypted = aws.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("kms_key_id"); ok {
+		request.KmsKeyId = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("role_name"); ok {
+		request.RoleName = aws.String(v.(string))
+	}
+
+	err := resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		var resp *ec2.ImportSnapshotOutput
+		resp, err := conn.ImportSnapshot(request)
+		// Error: InvalidParameter: The service role terraform-20201121225356951800000001 provided does not exist or does not have sufficient permissions
+		// status code: 400, request id: b0abc3d2-5b59-4e5c-b748-c1cb084020c0
+
+		if isAWSErr(err, "InvalidParameter", "provided does not exist or does not have sufficient permissions") {
+			return resource.RetryableError(err)
+		}
+
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		importTaskId := aws.StringValue(resp.ImportTaskId)
+
+		res, err := importAwsEbsSnapshotWaiter(conn, importTaskId)
+		if err != nil {
+			return resource.NonRetryableError(fmt.Errorf("Error waiting for snapshot (%s) to be imported: %s", d.Id(), err))
+		}
+
+		d.SetId(aws.StringValue(res.SnapshotId))
+
+		err = resourceAwsEbsSnapshotImportRead(d, meta)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+
+	if tfresource.TimedOut(err) {
+		return fmt.Errorf("timeout error importing EBS Snapshot: %s", err)
+	}
```

Comment:
This doesn't seem to be following the spirit of the rule, but I'm not confident in the proper way to handle this case.